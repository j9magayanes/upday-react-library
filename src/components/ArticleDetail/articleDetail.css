.articleAlign {
    position: relative;
    width: 100%;
    padding-top: 75%;
    overflow: hidden;
    border-radius: 32px;
}

.rightAlignment {
    position: absolute;
    inset: 0px;
    background-image: linear-gradient(rgba(0, 0, 0, 0.25), rgba(0, 0, 0, 0.6) 98.9%);
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    color: white;
}

.right {
    display: flex;
    flex-direction: column;
    justify-content: flex-end;
    margin: 24px;
    height: calc(100% - 48px);
}