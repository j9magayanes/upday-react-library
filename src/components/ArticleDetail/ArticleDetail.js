import React from 'react';
import './articleDetail.css';
import  'upday-web-components/components/articleDetail'

export const ArticleDetail = ({ className, previewText }) => {
  return (
    <article-detail data-testid="articleDetail" class={className} text={previewText}></article-detail>
  );
};