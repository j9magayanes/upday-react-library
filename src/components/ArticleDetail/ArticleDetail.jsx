import React from 'react';
import './articleDetail.css';


export const ArticleDetail = (   { primary, backgroundColor, size, label, title,  ...props }) => {
  return (
    <p id={props.id}  className={['storybook-title', `storybook-title--${size}`].join(' ')}>{title}</p>
  );
};