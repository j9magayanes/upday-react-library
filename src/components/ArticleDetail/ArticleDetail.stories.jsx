import React from 'react';

import { ArticleDetail } from './ArticleDetail';
// import { AdWrapper } from '../AdWrapper/AdWrapper'


export default {
  title: 'Form/ArticleDetail',
  component: ArticleDetail,
};

const Template = () =><ArticleDetail args={{items: [{name: "test", callback:"", href:""}]}}/>;


export const Default = Template.bind({});