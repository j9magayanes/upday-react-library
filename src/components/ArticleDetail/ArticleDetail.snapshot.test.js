import React from 'react';
import renderer from 'react-test-renderer';
import { ArticleDetail } from './ArticleDetail';

test('renders ArticleDetail component correctly', () => {
  const props = {
    className: 'your-class-name',
    previewText: 'Your preview text goes here.',
  };

  const component = renderer.create(<ArticleDetail {...props} />);
  const tree = component.toJSON();

  expect(tree).toMatchSnapshot();
});
