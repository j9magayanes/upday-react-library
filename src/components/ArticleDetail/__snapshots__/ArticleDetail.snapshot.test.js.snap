// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders ArticleDetail component correctly 1`] = `
<article-detail
  class="your-class-name"
  data-testid="articleDetail"
  text="Your preview text goes here."
/>
`;
